### PR TITLE
fix(public-cloud): revamp instance creation ui

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-4352
| License          | BSD 3-Clause

## Description

The Instance creation wizard has been revamped to ease the process of instance creation for the user, by reducing the number of steps the user has to perform

### :bookmark: TODO

- [x] TRANS-27236

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#12415.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12415.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12417.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12417.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12417.2 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12423.0 ✘  
* 
  * Build ✘

File: packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js [hunk 1]
```diff
@@ -65,6 +70,24 @@ export default class ImagesListController {
...
+    const os = {};
+    forEach(this.os, (distributions, imageType) => {
+      os[imageType] = {};
+      forEach(distributions, (images, distribution) => {
+        os[imageType][distribution] = filter(images, (image) =>
+          this.isCompatible(image),
+        );
+      });
+    });
+    this.filteredOs = os;
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js]: Good one. This is done :)

File: packages/manager/modules/pci/src/components/project/images-list/images-list.html [hunk 2]
```diff
@@ -2,13 +2,16 @@
...
+            data-model="$ctrl.showNonAvailable"
+            data-on-change="$ctrl.getDataToShow(modelValue)"
+        >
             <span data-translate="pci_project_instances_available_only"></span>
         </oui-checkbox>
 
         <oui-tabs class="mb-4" data-model="$ctrl.selectedTab">
             <oui-tabs-item
-                data-ng-repeat="(imageType, distributions) in $ctrl.os"
+                data-ng-repeat="(imageType, distributions) in $ctrl.osToShow"
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/images-list/images-list.html]: This is fixed

File: packages/manager/modules/pci/src/components/project/images-list/images-list.html [hunk 3]
```diff
@@ -20,57 +23,24 @@
...
-                            class="d-inline-block col-md-6 col-xl-4 my-3"
+                        <oui-select-picker
+                            class="d-inline-block col-md-6 col-xl-3 my-3"
+                            data-ng-repeat="(distribution, images) in distributions track by distribution"
+                            data-ng-if="$ctrl.showNonAvailable || $ctrl.isDistributionCompatible(images)"
+                            data-label="{{:: 'pci_project_instances_distribution_' + distribution | translate }}"
+                            data-variant="light"
+                            data-picture="{{$ctrl.IMAGE_ASSETS.os[distribution]}}"
+                            data-model="$ctrl.image"
+                            data-on-change="$ctrl.distribution = distribution; $ctrl.onImageChange(modelValue);"
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/images-list/images-list.html]: This is done

**Comment by ovh-cds** [comment]
CDS Report build#12432.0 ✘  
* 
  * Build ✘

File: packages/manager/modules/pci/src/components/project/images-list/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
-
-  "pci_project_instances_available_only": "Masquer les options non disponibles",
+  "pci_project_instances_available_only": "Montrer les localisations indisponibles",
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/components/project/images-list/translations/Messages_fr_FR.json:2]: nit: The key name should reflect as much as possible what it is written.
For example: pci_project_instances_display_nonavailable

File: packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js [hunk 5]
```diff
@@ -12,11 +13,13 @@ export default class ImagesListController {
   /* @ngInject */
   constructor(
```
Comment (radireddy) [inline packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js:14]: remove

File: packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js [hunk 6]
```diff
@@ -12,11 +13,13 @@ export default class ImagesListController {
...
     // $translate,
+    $q,
     OvhApiCloudProjectImage,
     OvhApiCloudProjectSnapshot,
     PciProjectImages,
   ) {
     // this.$translate = $translate;
```
Comment (radireddy) [inline packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js:21]: remove commented code

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html [hunk 7]
```diff
@@ -125,7 +104,7 @@
                             </div>
                         </oui-select-picker-section>
                         <oui-select-picker-section>
-                            <div class="text-right">
+                            <div>
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html]: This is done

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html [hunk 8]
```diff
@@ -185,49 +164,67 @@
...
-                            class="ml-auto"
-                            data-ng-if="!$ctrl.flavor.outboundBandwidth"
-                            >-</span
-                        >
+                    </div>
+                    <div
+                        class="d-flex justify-content-between"
+                        data-ng-if="$ctrl.flavor.getGpu()"
+                    >
+                        <span>
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html]: This is done

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html [hunk 9]
```diff
@@ -185,49 +164,67 @@
...
                     </div>
-                    <div class="d-flex justify-content-between">
-                        <span
-                            data-translate="pci_project_flavors_spec_bandwidth"
-                        ></span>
+                    <div
+                        class="d-flex justify-content-between"
+                        data-ng-if="$ctrl.flavor.getNvme()"
+                    >
+                        <span>
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html]: This is done

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html [hunk 10]
```diff
@@ -185,49 +164,67 @@
...
+                            </span>
+                            <span
+                                data-ng-bind="$ctrl.flavor.getGpuModel()"
+                            ></span>
+                        </span>
                     </div>
                 </oui-select-picker-section>
                 <oui-select-picker-section>
-                    <div class="text-right">
+                    <div>
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.html]: This is done

File: packages/manager/modules/pci/src/components/project/images-list/translations/Messages_fr_FR.json (hunk 4 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/components/project/images-list/translations/Messages_fr_FR.json]: This is done

**Comment by ganeshkumar1989** [comment]
The translations are done and are incorporated into the PR. Thanks!

**Pull request merged**